dynamic-tools/zap-extensions/zapwso2jiraplugin/LastScanReport.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied.  See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.zaproxy.zap.extension.zapwso2jiraplugin;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.extension.report.ReportGenerator;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteMap;
+import org.parosproxy.paros.model.SiteNode;
+import org.zaproxy.zap.extension.XmlReporterExtension;
+import org.zaproxy.zap.utils.XMLStringUtil;
+import org.zaproxy.zap.view.ScanPanel;
+
+/*
+* This class is used to create the report after a scan, Here methods are copied and modified from package
+* org.parosproxy.paros.extension.report; class ReportLastScan
+**/
+
+public class LastScanReport {
+
+    private static LastScanReport instance = new LastScanReport();
+
+    private LastScanReport() {
+    }
+
+    public static synchronized LastScanReport getInstance() {
```

Comment:
For singletons please use double-check locking with volatile instance variables:

Refer: http://stackoverflow.com/questions/7855700/why-is-volatile-used-in-this-example-of-double-checked-locking